From: David Paleino <d.paleino@gmail.com>
Subject: gracefully handle (and try to fix) configuration parsing errors
 From time to time, it happens that an empty section "[]" is added to
 wired-settings.conf. (Un)fortunately, RawConfigParser doesn't seem to
 remove it with self.remove_section(""), but it automagically disappears
 if the configfile is re-written again. Exceptions on errors are caught
 if that was not the error.
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=553197
 http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=554949
Forwarded: no

---
 wicd/configmanager.py |   15 ++++++++++++---
 1 file changed, 12 insertions(+), 3 deletions(-)

--- wicd.orig/wicd/configmanager.py
+++ wicd/wicd/configmanager.py
@@ -26,7 +26,7 @@ reusable for other purposes as well.
 
 import os, copy
 
-from ConfigParser import RawConfigParser
+from ConfigParser import RawConfigParser, ParsingError
 
 from wicd.misc import Noneify, to_unicode
 
@@ -39,8 +39,17 @@ class ConfigManager(RawConfigParser):
         self.config_file = path
         self.debug = debug
         self.mrk_ws = mark_whitespace
-        self.read(path)
-        
+        try:
+            self.read(path)
+        except ParsingError, e:
+            self.write()
+            try:
+                self.read(path)
+            except ParsingError, p:
+                import sys
+                print "Could not start wicd: %s" % p.message
+                sys.exit(1)
+
     def __repr__(self):
         return self.config_file
     
